### DISCUSSION TITLE
What should be the correct way to re-train images using Yolov3 darknet?

### DISCUSSION THREAD

**Discussion by bksharma13** [AlexeyAB/darknet#8619] [Q&A]
I have been scrolling the internet to read from the experts about custom training using Yolov3 via Darknet. There were great examples, but now I am at confusion on to perform how a retraining.

So here is my scenario -
I have some custom images with their class labels for training which I trained using a pre-trained model of yolov3.(yolov3-tiny.weights) When this training was completed, I renamed the final weight file as "yolov3-custom.weights".
Round 1 - Custom Classes=100, Max-batches = 100*2000 = 200000

Later on I have added new class labels and images and I again used the original tiny weight file.(yolov3-tiny.weights) During the training I saw that training time was way higher than the 1st round.
Round 2 - Custom Classes=160, Max-batches = 160*2000 = 320000

Similarly I am continuing now. But every-time my training time increases exponentially.

Now my question is -

1. Since there is no formal documents about re-training setup so I have no idea what should be my weight file in the 2nd rounds of training. Please recommend? yolov3-tiny.weights or yolov3-custom.weights

2. Let's assume, I was only adding new images to existing class labels in 2nd round of training and not adding or removing any class labels, which weight file should I have used? yolov3-tiny.weights or yolov3-custom.weights
And what should be Max-Batches as the weight file know internally the batches they have completed?
And also what is this type of training called - Fine training or Transfer Learning?

3. In 2nd round, if I was adding more images to existing class labels and also adding some new classes and performing a training, what will this be called - Fine training or Re-training?
If we used the custom weight then will it re-train on the new images?

Excuse me for asking multiple questions here. I have big time confusion.
